i3lock-pixeled: init at 1.1.0

Ma27 opened the pull request:
###### Motivation for this change

Simple derivation for my `i3lock` helper.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Ma27 committed:
i3lock-pixeled: init at 1.1.0

Ma27 commented:
I rebased onto the fixed master, so `travis-ci` should evaluate properly now %)

lsix commented on pkgs/misc/screensavers/i3lock-pixeled/default.nix:
```diff
@@ -0,0 +1,26 @@
+{ stdenv, pkgs, fetchurl }:
+
+stdenv.mkDerivation rec {
+  name = "i3lock-pixeled-pkg-${version}";
```
why is `pkg` in the name?

lsix commented on pkgs/misc/screensavers/i3lock-pixeled/default.nix:
```diff
@@ -0,0 +1,26 @@
+  version = "1.1.0";
+
+  src = fetchurl {
+    url = "https://github.com/Ma27/i3lock-pixeled/archive/1.1.0.tar.gz";
```
You could prefer:

```nix
    url = "https://github.com/Ma27/i3lock-pixeled/archive/${version}.tar.gz";
```

for easier maintenance.

Ma27 commented:
good catches, thanks @lsix

lsix commented on pkgs/misc/screensavers/i3lock-pixeled/default.nix:
```diff
@@ -0,0 +1,26 @@
+    sha256 = "046qbx4qvcc66h53h4mm9pyjj9gjc6dzy38a0f0jc5a84xbivh7k";
+  };
+
+  buildInputs = with pkgs; [ i3lock imagemagick scrot playerctl ];
```
I think that `buildInputs` should be renamed `propagatedBuildInputs` since all those programs are required at run-time and not present otherwise.

sifmelcara commented:
The script cannot find `scrot`, `convert`, and `i3lock` at runtime if these programs are not installed in user's environment. (And I believe `propagatedBuildInputs` do not help solve this problem)
I would probably use a wrapper and add necessary binaries to `$PATH` before the execution of i3lock-pixeled. (Not sure if this is the best solution)

Ma27 commented:
I added a patchPhase (inspired by `i3lock-fancy` derivation) which replaces all executables with fully-qualified nix-store paths.

Furthermroe I squashed everything into a single commit to avoid a bloated history.

Ma27 commented:
anything else missing??

Ma27 commented:
I rebased onto the current master and the build seems fine now...

Mic92 committed:
i3lock-pixeled: use substituteInPlace

Ma27 commented on pkgs/misc/screensavers/i3lock-pixeled/default.nix:
```diff
@@ -0,0 +1,39 @@
+       --replace i3lock    "${pkgs.i3lock}/bin/i3lock" \
+       --replace convert   "${pkgs.imagemagick}/bin/convert" \
+       --replace scrot     "${pkgs.scrot}/bin/scrot" \
+       --replace playerctl "${pkgs.playerctl}/bin/playerctl#"
```
I think that this breaks the executable as `#` is interpreted as comment in a shell file, right?

Mic92 replied:
oh. typo

Mic92 replied:
fixed in 56835b1eeaa420b40917e8f5303ee834bf8fa098

Ma27 replied:
ok, thanks :-)
